retry.go

```diff
@@ -0,0 +1,62 @@
+package rueidis
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/redis/rueidis/internal/util"
+)
+
+const (
+	defaultMaxRetryDelay = 1 * time.Second
+	maxAttemptsShift     = 63 // Avoid excessive shifts
+)
+
+// RetryDelay returns the delay that should be used before retrying the
+// attempt. Will return negative delay if the delay could not be determined or do not retry.
+type RetryDelay func(attempts int, err error) time.Duration
+
+// defaultRetryDelay delays the next retry exponentially without considering the error.
+// max delay is 1 second.
+func defaultRetryDelay(attempts int, _ error) time.Duration {
+	base := util.FastRandFloat64()
+	backoff := uint64(1 << uint64(math.Min(float64(attempts), maxAttemptsShift)))
+	jitter := base * float64(backoff) * float64(time.Millisecond)
+	return time.Duration(math.Min(jitter, float64(defaultMaxRetryDelay)))
```

Comment:
We should apply randomness only to the jitter part. The delay produced by the current implementation is not monotonically increasing.

Can we avoid using float64, including the `base`? We can use the built-in `min` and `max` functions.

Reply:
For example:

```go
	backoff := time.Microsecond * 5 * (1 << min(20, attempts))
	jitter := time.Microsecond * time.Duration(rand.Int64N(1<<min(20, attempts)))
	return min(time.Second, backoff+jitter)
```

Reply:
Well, actually above logic came from [here](https://aws.amazon.com/ko/blogs/architecture/exponential-backoff-and-jitter/).
> The delay produced by the current implementation is not monotonically increasing.

Intended part. To smooth multiple client calls.

Reply:
[d172784](https://github.com/redis/rueidis/pull/643/commits/d172784d4127a8955dcfe1ca9c3caf586fba7872)